tests/test_license.py

```diff
@@ -0,0 +1,25 @@
+"""
+Tests of the ``license`` field in ``pyproject.toml``, which can be populated
+from setuptools' ``license`` field.
+"""
+
+# setuptools *ONLY* supports bare license strings in the `license` field.  There exists a
+# `license_files` and a deprecated `license_file` field.  The latter simply raises a warning and the
+# content is prepended to `license_files`.  Neither populates the `license` property.
+# `license_files` is the subject of a PEP: https://peps.python.org/pep-0639/
+
+
+def test_license_setupcfg_string(project) -> None:
+    license_string = "Mock License v1.0"
```

Comment:
I think it can be either, if a "standard" license fits, you use the license name, otherwise if it's something special you spell out what that special license is.

https://packaging.python.org/en/latest/specifications/core-metadata/#license

i.e.
```python
license_string = """This software may only be obtained by sending the
author a postcard, and then the user promises not
to redistribute it."""
```

or

```python
license_text="GPL version 3, excluding DRM provisions"
```

`MIT` was chosen since it matches the classifiers used elsewhere in the tests.

Reply:
Ah I see, thanks for looking that up.

All I want to head off is that, if someone happens to be looking at our code and tests as an example of what to put in the `license` field of their `setup.cfg` (it could happen :shrug:), and in every case they see something that looks like the name of a license, they might come away with the impression that the field is _supposed_ to contain the name of the license and not something else. Yes, it's technically better to read the official documentation but I couldn't realistically fault anyone for looking at an example and copying its style - certainly I've done that often enough myself.

Reply:
Yep… well I've put tests that check both ways actually work as expected.  Since both ways are supported upstream.

PEP-639 should clean this up though.  The core metadata will have a `License-Expression` field (stored as `license` in the TOML) that'll be a SPDX license expression.  So in the future, we will need to validate whether this is a valid SPDX license expression, and if it is, it goes into `license`, otherwise, we write it out to a file and reference it in `license_files`.

So the current `license` "table" will change if PEP-639 is ratified in the form it takes today.

Reply:
...I suppose that works. I mean, I wasn't claiming that it actually matters for the tests, because as far as the test infrastructure is concerned, they're both just strings, but I guess it doesn't hurt to have a test with a short string and a long string just in case of some extremely obscure error that would affect one case and not the other, and having both cases in the tests certainly does give them more value as "documentation".

And yeah, things will look cleaner after PEP 639 but we can deal with that as it comes up.